### DISCUSSION TITLE
Better logs experience

### DISCUSSION THREAD

**Discussion by danielo515** [LunarVim/LunarVim#3202] [Ideas]
### Feature Description

Logs are, sadly, underrated. People just write to them when they faced a problem and want to debug it in the future, or when they "feel like this should be to the log". Also, normal users only think about them when things fail, so in general nobody gives logs much love unless things go wrong.
Current log experience for lunar-vim is a bit inconsistent, the log is lacking detail and the different things that can go wrong are spread across several places, and I hope that to change.
I have my log level set to debug, and I still get very few information about what is going on. If I open the lunarvim log after opening the editor, usually I see nothing new, just packer compile and those things seems to end up in the log.
Also, if something fails during runtime and it is properly caught and handled, usually those problems don't end up on the log either. They appear as a toast/notification and they end there.
Here is my proposal to make this experience richer and more useful when debugging problems. Not sure how feasible/desirable each thing is, but I want to propose it anyway.

This is my improvement plan:
- Notifications should also be sent to the log. I don't now if notify can be configured to also send stuff to a logging interface, but if it is possible I strongly thing it should, if not some wrapper or manual work may be required
- Initialization process should be logged, probably under the debug level, but I still think it should be logged. Ideally, each unit of configuration should log an start and a finish event (plugin install, template installation, reading user config, mapping key bindings, setting up commands, etc). In any case "Initialization complete" or something like that should always be logged, so when you open the log you can have a clear start point for the logs of the current session.
- Provide an already initialized log utility for the users and document it. This way people can log whatever they think makes sense within their config lua files and get all that useful information in a single centralized place
- Write some docs about logging. It is still confusing to me what each log is for. Should I be checking general neovim log? Should I check lunarvim log? or maybe I should open packer log? In all the cases, when I open them I usually find less information than I expect. I think that lunar vim log should be clearly documented: what is logged to it, what is not logged to it and should be checked in the other two logs, etc.

Regards

### Describe the alternatives you have considered

There are not much an user can do if the tool they use does not log enough information

[a comment by an automated account is not shown]

**Comment by danielo515** [comment]
@lvimuser @kylo252 should we open this as a discussion and try to came up with a concrete fix to open as an issue?

**Comment by danielo515** [comment]
@kylo252 can you provide some comments here?
My main problems with current login are:
- There is no documentation, so developers and users can not use it properly
- Many of the log calls appear as notify entries and don't end up in the log
- Many log calls do not even end up on the log
Regards

**Reply by kylo252** [reply]
@danielo515, please share your thoughts on #3294

**Comment by danielo515** [comment]
@kylo252 I already discussed there, but I think it has little to do with this specific "problem"

**Reply by kylo252** [reply]
it actually is related, because now it's easier to test this option
https://github.com/LunarVim/LunarVim/blob/36c8bdee9ff59a0a63c1edfc445b5eb2886cf246/lua/lvim/config/defaults.lua#L41-L42

I noticed that we're calling the wrong function here, please open a PR to fix it since it should solve the first point :) 
```diff
diff --git a/lua/lvim/core/log.lua b/lua/lvim/core/log.lua
index d88a659..7b95103 100644
--- a/lua/lvim/core/log.lua
+++ b/lua/lvim/core/log.lua
@@ -88,7 +88,7 @@ function Log:init()
         vim_log_level = vim_log_level + 1
       end
 
-      self:info(vim_log_level, msg)
+      self:add_entry(vim_log_level, msg)
     end
   end
```
